Add constant deprecation blog post (#2018)
home-assistant/developers.home-assistant: blog/2023-12-19-constant-deprecation.md

---
author: Robert Resch
authorURL: https://github.com/edenhaus
title: "Deprecated constants create a log entry"
---

In recent years many constants have been replaced by Enums or other values by adding a code comment pointing to the successor.
Using deprecated constants will now create a warning log entry containing the used integration, the replacement, and the version where the constant will be removed from core.

> We use a one-year deprecation period to ensure all custom integration authors have time to adjust.

Reviewer: ```suggestion
There will be a one-year deprecation period to ensure all custom integration authors have time to adjust.
```

> Most constants should not be used anymore, as we created several blog posts in the past about their deprecation. Some are listed below:

Reviewer: ```suggestion
Most constants should already have been replaced, as we created several blog posts in the past about their deprecation. Some are listed below:
```
